### PR TITLE
Bridge to ctl native interval

### PR DESCRIPTION
**Pull request by bladyjoker**
Related to #20

### CODE REVIEW

**Comment by bladyjoker** [comment]
```
purescript-bridge-typelib-spago-build> Error found:                                                                                                                                      
purescript-bridge-typelib-spago-build> in module Plutus.V1.Ledger.Contexts                                                                                                               
purescript-bridge-typelib-spago-build> at src/hgrb3hkgqgqlm5rgaph43fvwshicsx7r-generate-plutus-ledger-api-typelib/Plutus/V1/Ledger/Contexts.purs:53:14 - 53:29 (line 53, column 14 - line
 53, column 29)                                                                                                                                                                          
purescript-bridge-typelib-spago-build>   No type class instance was found for                                                                                                            
purescript-bridge-typelib-spago-build>                                                                                                                                                   
purescript-bridge-typelib-spago-build>     ToData.ToData (Interval POSIXTime)                                                                                                            
purescript-bridge-typelib-spago-build>                                                                                                                                                   
purescript-bridge-typelib-spago-build> while solving type class constraint                                                                                                               
purescript-bridge-typelib-spago-build>                                                                                                                                                   
                                                                                                                                                                                         
purescript-bridge-typelib-spago-build>   ToData.ToDataArgsRL (Cons @Type "txInfoValidRange" (Interval POSIXTime) (Cons @Type "txInfoWdrl" (Array (Tuple StakingCredential BigInt)) (Nil @
Type)))                                                                                                                                                                                  
purescript-bridge-typelib-spago-build>                       t2                                                                                                                          
                                                                                                                                                                                         
purescript-bridge-typelib-spago-build>                                                                                                                                                   
                                                                                                                                                                                         
purescript-bridge-typelib-spago-build> while applying a function genericToData                                                                                                           
purescript-bridge-typelib-spago-build>   of type Generic t0 t1 => ToDataWithIndex t1 t0 => t0 -> PlutusData                                                                              
purescript-bridge-typelib-spago-build>   to argument x                                                                                                                                   
purescript-bridge-typelib-spago-build> while checking that expression genericToData x                                                                                                    
purescript-bridge-typelib-spago-build>   has type PlutusData                                                                                                                             
purescript-bridge-typelib-spago-build> in value declaration $toDataTxInfo23                                                                                                              
purescript-bridge-typelib-spago-build> where t0 is an unknown type                                                                                                                       
purescript-bridge-typelib-spago-build>       t1 is an unknown type                                                                                                                       
purescript-bridge-typelib-spago-build>       t2 is an unknown type                                                                                                                       
purescript-bridge-typelib-spago-build> See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,                                       
purescript-bridge-typelib-spago-build> or to contribute content related to this error.                                                                                                   
```

We need to specify ToData/FromData instances in CTL for this to work, ugh this is a pain

File: src/PlutusTx/LedgerTypes.hs [hunk 1]
```diff
@@ -84,16 +87,19 @@ writeLedgerTypesAnd :: FilePath -> [SumType 'Haskell] -> IO ()
 writeLedgerTypesAnd fp myTypes =
   writePSTypes
     fp
-    (buildBridge (defaultBridge <|> origToCtlNativePrimitivesBridge <|> origToCtlNativeOverriddenBridge))
+    (buildBridge plutusLedgerApiBridge)
     (ledgerTypes <> myTypes)
 
+plutusLedgerApiBridge :: BridgeBuilder PSType
+plutusLedgerApiBridge = defaultBridge <|> origToCtlNativePrimitivesBridge <|> origToCtlNativeOverriddenBridge <|> origToCtlNativeScriptsBridge
```
Comment (MangoIV) [inline src/PlutusTx/LedgerTypes.hs:94]: just a dumb question: what is the intent behind the choice? Sooner is better? Or are those disjoint?

Comment (bladyjoker) [reply src/PlutusTx/LedgerTypes.hs:94]: not a dumb question, they should be disjoint

Comment (bladyjoker) [reply src/PlutusTx/LedgerTypes.hs:94]: it's mostly to organize the code and to establish a vocab when talking about what gets bridge where

**Comment by bladyjoker** [comment]
@MangoIV https://github.com/mlabs-haskell/purescript-bridge/blob/master/nix/purescript-bridge-typelib-spago/packages.dhall#L176 is where the CTL package is updated. Very hidden X)

File: src/PlutusTx/LedgerTypes.hs (hunk 1 above)
Comment (MangoIV) [reply src/PlutusTx/LedgerTypes.hs:94]: Thank you for clarifying! :)

**Comment by bladyjoker** [comment]
Done by #21

**Pull request closed without merging**